src/test/java/com/endava/internship/mocking/repository/InMemPaymentRepositoryTest.java

```diff
@@ -0,0 +1,122 @@
+package com.endava.internship.mocking.repository;
+
+import com.endava.internship.mocking.model.Payment;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatExceptionOfType;
+import static org.junit.jupiter.api.Assertions.*;
+
+class InMemPaymentRepositoryTest {
+
+    private PaymentRepository underTest;
+    private Payment payment1;
+    private Payment payment2;
+    private Payment payment3;
+
+    @BeforeEach
+    public void setUp() {
+        underTest = new InMemPaymentRepository();
+        payment1 = new Payment(1, 100.5, "Payment No. 1");
+        payment2 = new Payment(2, 350.65, "Payment No.2");
+        payment3 = new Payment(3, 500.30, "Payment No. 3");
```

Comment:
would recommend to initialize there as class fields and can be declared as static.